engines/hypno/spider/spider.cpp

```diff
@@ -0,0 +1,192 @@
+	_levels[arclevel].arcade.levelIfWin = "sixdemo/mis/demo.mis";
+	_levels[arclevel].arcade.levelIfLose = "sixdemo/mis/demo.mis";
+
+	loadLib("", "sixdemo/c_misc/fonts.lib");
+	loadLib("sixdemo/c_misc/sound.lib/", "sixdemo/c_misc/sound.lib");
+	loadLib("sixdemo/demo/sound.lib/", "sixdemo/demo/sound.lib");
+
+	// Read assets from mis files
+	parseScene("sixdemo", "mis/demo.mis");
+	ChangeLevel *cl = new ChangeLevel();
+	cl->level = "sixdemo/c_misc/missions.lib/c1.mi_";
+	_levels["sixdemo/mis/demo.mis"].scene.hots[1].actions.push_back(cl);
+
+	cl = new ChangeLevel();
+	cl->level = "sixdemo/mis/alley.mis";
+	_levels["sixdemo/mis/demo.mis"].scene.hots[2].actions.push_back(cl);
+
+	cl = new ChangeLevel();
+	cl->level = "sixdemo/puz_matr";
+	_levels["sixdemo/mis/demo.mis"].scene.hots[3].actions.push_back(cl);
+
+	cl = new ChangeLevel();
+	cl->level = "sixdemo/mis/shoctalk.mis";
+	_levels["sixdemo/mis/demo.mis"].scene.hots[4].actions.push_back(cl);
+
+	cl = new ChangeLevel();
+	cl->level = "sixdemo/mis/order.mis";
+	_levels["sixdemo/mis/demo.mis"].scene.hots[5].actions.push_back(cl);
+	_levels["sixdemo/mis/demo.mis"].scene.sound = "demo/sound.lib/menu_mus.raw";
+
+	parseScene("sixdemo", "mis/order.mis");
+	cl = new ChangeLevel();
+	cl->level = "<quit>";
+	_levels["sixdemo/mis/order.mis"].scene.hots[1].actions.push_back(cl);
+
+	parseScene("sixdemo", "mis/alley.mis");
+	_levels["sixdemo/mis/alley.mis"].scene.intro = "demo/aleyc01s.smk";
+	_levels["sixdemo/mis/alley.mis"].scene.sound = "demo/sound.lib/alleymus.raw";
+	_levels["sixdemo/mis/alley.mis"].scene.levelIfWin = "sixdemo/mis/demo.mis";
+	_levels["sixdemo/mis/alley.mis"].scene.levelIfLose = "sixdemo/mis/demo.mis";
+
+	parseScene("sixdemo", "mis/shoctalk.mis");
+
+	Level matrix;
+	matrix.puzzle.name = "sixdemo/puz_matr";
+	matrix.puzzle.intros.push_back("spiderman/demo/aleyc01s.smk");
+	matrix.puzzle.levelIfWin = "sixdemo/mis/demo.mis";
+	matrix.puzzle.levelIfLose = "sixdemo/mis/demo.mis";
+	_levels["sixdemo/puz_matr"] = matrix;
+	_soundPath = "c_misc/sound.lib/";
+}
+
+void SpiderEngine::runPuzzle(Puzzle puzzle) {
+	if (puzzle.name == "sixdemo/puz_matr")
+		runMatrix(puzzle);
+	else
+		error("invalid puzzle");
+}
+
+void SpiderEngine::runMatrix(Puzzle puzzle) {
+	Common::Point mousePos;
+	Common::Event event;
+
+	defaultCursor();
+	bool data[10][10] = {};
+	bool solution[10][10] = {
+		{0, 0, 0, 1, 1, 1, 1, 0, 0, 0},
+		{0, 0, 1, 1, 1, 1, 1, 1, 0, 0},
+		{0, 1, 1, 1, 1, 1, 1, 1, 1, 0},
+		{0, 1, 1, 0, 0, 0, 0, 1, 1, 0},
+		{0, 1, 1, 1, 1, 1, 1, 1, 1, 0},
+		{0, 0, 1, 1, 1, 1, 1, 1, 0, 0},
+		{0, 0, 1, 1, 0, 0, 1, 1, 0, 0},
+		{0, 0, 1, 1, 0, 0, 1, 1, 0, 0},
+		{0, 0, 0, 1, 0, 0, 1, 0, 0, 0},
+		{0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+	};
+	Common::Rect matrix(175, 96, 461, 385);
+	Common::Rect cell(0, 0, 27, 27);
+	uint32 activeColor = _pixelFormat.RGBToColor(0, 130, 0);
+	uint32 deactiveColor = _pixelFormat.RGBToColor(0, 0, 0);
+	int x, y;
```

Comment:
please reduce the scope of these variables.

Reply:
Done.